EIPS/eip-6806.md

```diff
@@ -0,0 +1,98 @@
+---
+eip: 6806
+title: ERC-721 Holding Time Tracking
+description: Add holding time information to ERC-721 tokens
+author: Saitama (@saitama2009), Combo <combo@1combo.io>, Luigi <luigi@1combo.io>
+discussions-to: https://ethereum-magicians.org/t/draft-eip-erc721-holding-time-tracking/13605
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-03-30
+requires: 721
+---
+
+## Abstract
+
+This standard is an extension of [ERC-721](./eip-721.md). It adds an interface that tracks and describes the holding time of a Non-Fungible Token (NFT) by an account. 
+
+## Motivation
+
+In some use cases, it is valuable to know the duration for which a NFT has been held by an account. This information can be useful for rewarding long-term holders, determining access to exclusive content, or even implementing specific business logic based on holding time. However, the current ERC-721 standard does not have a built-in mechanism to track NFT holding time. Furthermore, in certain scenarios, the holding time should not be reset when an NFT is transferred between accounts, such as during NFT-based staking.
+
+This proposal aims to address these limitations by extending the ERC-721 standard to include holding time tracking functionality and allowing for selective exclusion of transfers from affecting holding time calculations.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+**Interface**
+
+The following interface extends the existing ERC-721 standard:
+
+```solidity
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.0
+
+interface IERC721HoldingTime {
+  function getHoldingInfo(uint256 tokenId) external view returns (address holder, uint256 holdingTime);
+  function setWhitelistedContract(address whitelistContract, bool ignoreReset) external;
```

Comment:
Do you want to enable an event to be emitted when contract is being allowlisted? 
Also, can you change the `setWhitelistedContract` to more specific to ERC721HoldingTime since this method name is not very obvious related to holding time

Does it have to be a "contract" or it could be a general address of EOA/Contract?